metadatastore/commands.py

```diff
@@ -790,25 +1106,34 @@ def find_events(descriptor=None, **kwargs):
     if 'event_descriptor_id' in kwargs:
         raise ValueError("Use 'descriptor_id', not 'event_descriptor_id'.")
 
-    _format_time(kwargs)
-    # get the actual mongo document
     if descriptor:
-        descriptor = _get_mongo_document(descriptor, EventDescriptor)
-        kwargs['descriptor_id'] = descriptor.id
+        descriptor = doc_or_uid_to_uid(descriptor)
+
+        descriptor = descriptor_given_uid(descriptor)
+        descriptor = _EVENTDESC_UID_to_OID_MAP[descriptor['uid']]
+        kwargs['descriptor_id'] = descriptor
+
+    _format_time(kwargs)
 
-    _normalize_object_id(kwargs, '_id')
-    events = Event.objects(__raw__=kwargs).order_by('-time')
+    events = Event.objects(__raw__=kwargs).order_by('time')
     events = events.as_pymongo()
-    dref_dict = dict()
-    name = Event.__name__
-    for n, f in Event._fields.items():
-        if isinstance(f, ReferenceField):
-            lookup_name = f.db_field
-            dref_dict[lookup_name] = f
 
-    _as_document = _AsDocumentRaw()
-    return (reorganize_event(_as_document(name, ev, dref_dict))
-            for ev in events)
+    for ev in events:
+        # ditch the ObjectID
+        ev.pop('_id')
+        # pop the descriptor oid
+        desc_oid = ev.pop('descriptor_id')
+        # replace it with the defererenced descriptor
+        ev['descriptor'] = _descriptor_given_oid(desc_oid)
```

Comment:
could probably make this `ev['descriptor'] = _descriptor_given_oid(ev.pop('descriptor_id'))` and delete the line where you capture `desc_oid`